CraftsmenLtd/BloodConnect
# I#254: Added phone number input for SSO (Google/Facebook) login flow

Fixes #254


## What kind of change does this PR introduce (check at least one)


## How would you describe the changes in a changelog (1-2 lines)?

Added a phone number field to the `AddPersonalInfo` page that only appears for SSO (Google/Facebook) login users, since they haven't provided their phone number yet. Regular email registration users won't see this field since they already entered their phone number during signup.

## How to test

Testing steps:
1. Login using Google/Facebook SSO
2. Verify phone number input field appears in AddPersonalInfo page
3. Try submitting without phone number - should show validation error
4. Enter invalid phone number - should show format validation error
5. Enter valid phone number - should save successfully
6. Verify phone number is saved in DynamoDB
7. Login using manual registration - verify phone number field does NOT appear in AddPersonalInfo

## Checklist


## Additional Notes

- The phone number validation uses the same format as manual registration
- Phone numbers are stored in the same DynamoDB field as manual registration
- Made the implementation conditional so it only appears for SSO users

## Deploy risk & follow up


Rationale: This is an additive change that only affects SSO users during registration. The phone number handling code reuses existing validation and storage logic from manual registration. The changes are isolated to the `AddPersonalInfo` component and don't affect other parts of the application.
